dpppt-backend-sdk/dpppt-backend-sdk-ws/src/test/java/org/dpppt/backend/sdk/ws/controller/GaenControllerTest.java

```diff
@@ -893,30 +886,26 @@ public void testTokenValiditySurpassesMaxJwtValidity() throws Exception{
 		int maxJWTValidityInMinutes = 60;
 		String token = createToken(now.plusMinutes(maxJWTValidityInMinutes + 1));
 
-		mockMvc.perform(post("/v1/gaen/exposed")
-						.contentType(MediaType.APPLICATION_JSON).header("Authorization", "Bearer " + token)
-						.header("User-Agent", "MockMVC").content(json(exposeeRequest))).andExpect(status().is(401));
-}
+		mockMvc.perform(post("/v1/gaen/exposed").contentType(MediaType.APPLICATION_JSON)
+				.header("Authorization", "Bearer " + token).header("User-Agent", "MockMVC")
+				.content(json(exposeeRequest))).andExpect(status().is(401));
+	}
 
 	@Test
 	public void testDebugController() throws Exception {
 		var now = UTCInstant.now();
 		var midnight = now.atStartOfDay();
 
-		insertNKeysPerDayInIntervalWithDebugFlag(14,
-				midnight.minusDays(4),
-				midnight, midnight.minusDays(1), true);
+		// insert two times 10 keys per day for the last 14 days, with different received at. In total: 280 keys
+		insertNKeysPerDay(midnight, 14, 10, midnight.minusDays(1), true);
+		insertNKeysPerDay(midnight, 14, 10, midnight.minusHours(12), true);
 
-		insertNKeysPerDayInIntervalWithDebugFlag(14,
-				midnight.minusDays(4),
-				midnight, midnight.minusHours(12), true);
+		
+		// request keys which have been received in the last day, must be 280 in total.
 		MockHttpServletResponse response = mockMvc
-				.perform(get("/v1/debug/exposed/"
-						+ midnight.getTimestamp())
-								.header("User-Agent", "MockMVC"))
+				.perform(get("/v1/debug/exposed/" + midnight.getTimestamp()).header("User-Agent", "MockMVC"))
 				.andExpect(status().is2xxSuccessful()).andReturn().getResponse();
-
-		verifyZipInZipResponse(response, 0);
+		verifyZipInZipResponse(response, 280, 144);
```

Comment:
Can you please explain to me why the system returns all 280 keys here? I would've thought it returns 
- 10 random
- 10 from the first `insertNKeysPerDay`
- 10 from the second `insertNKeysPerDay`
= 30?

Reply:
See additional comment in code.